solana/program_v2/programs/gateway_v2/src/state/gatekeeper.rs

```diff
@@ -313,3 +299,421 @@ bitflags! {
 impl OnChainSize for GatekeeperKeyFlags {
     const ON_CHAIN_SIZE: usize = OC_SIZE_U32;
 }
+
+#[cfg(test)]
+mod gatekeeper_tests {
+    use super::*;
+    use crate::errors::GatekeeperErrors;
+    use crate::instructions::network::UpdateGatekeeperFees;
+    use solana_program::clock::Epoch;
+
+    #[test]
+    fn test_can_access_with_auth_authority() {
+        let key = &Pubkey::new_unique();
+        let owner = &Pubkey::new_unique();
+        let data = &mut vec![1, 2, 3];
+        let lamports = &mut 0;
+        let account_info = AccountInfo::new(
+            key,
+            true,
+            true,
+            lamports,
+            data,
+            owner,
+            false,
+            Epoch::default(),
+        );
+
+        let authority = Signer::try_from(&account_info).unwrap();
```

Comment:
Struggling to make this int a helper function due to ownership issues. Would love some help here

Reply:
@tbarri helped with a great little function!